src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -2313,7 +2326,8 @@ bool StorageReplicatedMergeTree::executeLogEntry(LogEntry & entry)
     {
         ProfileEventsScope profile_events_scope;
 
-        if (MutableDataPartPtr part = attachPartHelperFoundValidPart(entry))
+        PartsTemporaryRename renamed_parts(*this, DETACHED_DIR_NAME);
```

Comment:
`PartsTemporaryRename` needs here to move part back (`/detached/attacing_<part_name>` -> `/detached/<part_name>`) in case committing part to the active set is failed.